### DISCUSSION TITLE
Desktop extra keyboard shortcuts

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#309] [Q&A] [answered]
Cruiser desktop keyboard shortcuts 

Push to HIDE. 
I: turn instructions
S: shaping points
V: via points
W: shaping + via points

**Comment by devemux86** [comment]
Next desktop version will support extra keyboard shortcuts to show / hide:
- `I`: turn instructions
- `S`: shaping points
- `V`: via points
- `W`: shaping + via points

Easy to implement because the support in the code for these already exists.

**Reply by devemux86** [reply]
Desktop **[Cruiser 5.1.0-b11](https://github.com/devemux86/cruiser/discussions/671)** was updated with the new keyboard shortcuts.

**Reply by devemux86** [reply, marked as answer]
- New keyboard shortcuts are available in desktop https://github.com/devemux86/cruiser/discussions/684

**Comment by Basti32** [comment]
Hallo @devemux86,
da ich gerne mit der App Routen plane, mit der ich auch navigiere, habe ich mir Blue Stracks AppPlayer (Androidsimulator) auf dem Rechner installiert und kann so im Cruiser meine Routen planen. 
Nun lese ich hier von Tastenkombinationen. Sollten diese auf meinem Rechner so funktionieren oder ist hier etwas anderes gemeint mit Desktop Cruiser?
Würde mich sehr freuen, wenn mir da einer weiterhelfen könnte.
Viele Grüße
Basti

Transaltor DeepL:

Hello @devemux86, 
as I like to plan routes with the app that I also use to navigate, I have installed Blue Strack's AppPlayer (Android simulator) on my computer and can plan my routes in the Cruiser. 
Now I read here about key combinations. Should these work like this on my computer or is something else meant here with Desktop Cruiser? I would be very happy if someone could help me further. 
best regards 
Basti

**Reply by devemux86** [reply]
@Basti32 

These keyboard shortcuts only work on the desktop Cruiser, not on Android.

Diese Tastaturkürzel funktionieren nur auf dem Desktop-Cruiser, nicht auf Android.

**Reply by bios4** [reply]
@Basti32 you can get the Cruiser app for desktop here: https://github.com/devemux86/cruiser/releases (download the cruiser*.zip file)

No need for an "Android emulator" or anything like this.
I think to run it you "just" need a suitable Java runtime environment...

**Comment by Basti32** [comment]
Hallo @devemux86,
danke für die Antwort. Ich habe mir schon so etwas gedacht. Somit nützt es mir nichts, wenn ich mir für mein Tab eine Tastatur zulege.

Hallo @bios4,
danke für den Link. Werde ich mir mal anschauen und vielleicht ist das die Alternative für meine bis jetzige Routenplanung.

Viele Grüße
Basti

Transalator Deepl

Hi @devemux86, 
thanks for the reply. I was already thinking something like that. So there's no point in getting a keyboard for my Tab. 

Hi @bios4, 
thanks for the link. I'll have a look and maybe that's the alternative for my current route planning. 
Best regards 
Basti

**Reply by devemux86** [reply]
@Basti32

Route planning with curvy routes and different route profiles for sections is only available in Android Cruiser.

Die Routenplanung mit kurvenreichen Strecken und verschiedenen Streckenprofilen für Abschnitte ist nur in Android Cruiser verfügbar.

**Reply by WolfgangZ42** [reply]
> So there's no point in getting a keyboard for my Tab.

That's of course your decision - but believe me, it's easier to plan/change routes with a keyboard and a mouse! 

![2024-12-23 12 03 51](https://github.com/user-attachments/assets/b8a6315c-7a95-43a5-aad5-7e80eb91cc35)
My equipment - I wont miss it. You can easily take it with you on the tour, but even at home it's better as kurviger.com if you are dealing with tracks (TET/ACT/MDmod).

**Comment by Basti32** [comment]
Hallo @bios4, 
in der Datei Cruiser finde ich leider keine EXE oder eine Installationsdatei. Java habe ich installiert. Komme ab hier nicht weiter.

Hallo @devemux86 ,
> Die Routenplanung mit kurvenreichen Strecken und verschiedenen Streckenprofilen für Abschnitte ist nur in Android Cruiser verfügbar.

das ist nicht weiter schlimm, da ich meine Routen 80 % selber abstecke.

Hallo @0709wiwiw,
mit TET und Co liegen sie schon mal na dran. Wobei ich diese Tracks meist nur als Vorschlag nehme und immer wieder davon abweiche. 
So sieht Desktop-Cruiser aus?

Viele Grüße
Basti

Transaltor Deepl

Hello @bios4, 
unfortunately I cannot find an EXE or an installation file in the Cruiser file. I have installed Java. 

Hello @devemux86 , 

> Route planning with curvy routes and different route profiles for sections is only available in Android Cruiser.

That's not too bad, because I stake out 80% of my routes myself. 

Hello @0709wiwiw, 
with TET and Co you are already close to it. However, I usually only use these tracks as a suggestion and always deviate from them. 
This is what desktop cruisers look like? 
Best regards 
Basti

**Reply by devemux86** [reply]
@Basti32 

https://wiki.openstreetmap.org/wiki/Cruiser#Installation

- **Windows**: A working Java Runtime Environment installation is required. Download the latest Cruiser zip. Unpack the archive and run the **cruiser.bat**.

- **macOS / Linux**: A working Java Runtime Environment installation is required. Download the latest Cruiser zip. Unpack the archive and run the **cruiser.sh**.

---

- **Windows**: Eine funktionierende Java Runtime Environment-Installation ist erforderlich. Laden Sie die neueste Cruiser-Zip-Datei herunter. Entpacken Sie das Archiv und führen Sie die Datei **cruiser.bat** aus.

- **macOS / Linux**: Eine funktionierende Java Runtime Environment-Installation ist erforderlich. Laden Sie die neueste Cruiser-Zip-Datei herunter. Entpacken Sie das Archiv und führen Sie die Datei **cruiser.sh** aus.

**Reply by Basti32** [reply]
Hallo @devemux86,
danke für die Hilfe, Jetzt funktioniert es. Dann werde ich mich mal bei Desktop-Cruiser umschauen. 
Viele Grüße
Basti

**Comment by Basti32** [comment]
Hallo,
dank für die Hilfe. Desktop-Cruiser läuft jetzt auch. Bei dem Tastenkürzel geht aber nur I, G und H. Bei dem Tastenkürzel H wird die Karte scheinbar neu geladen.
Bei allen anderen sehe ich keine Reaktion.
Muss ich noch etwas beachten?

Viele Grüße
Basti

Transaltor Deppl:

Hello, 
thanks for the help. Desktop Cruiser is now also running. However, only I, G and H work with the keyboard shortcut. The map seems to be reloaded with the keyboard shortcut H. I don't see any reaction with any of the others. Is there anything else I need to consider? 
Best regards 
Basti

**Reply by devemux86** [reply]
@Basti32 

You can move the map with the arrow keys and zoom with `+`/`-`.
`H` turns on / off the hillshading.
The available keyboard shortcuts are mentioned above:

Sie können die Karte mit den Pfeiltasten verschieben und mit `+`/`-` zoomen.
`H` schaltet die Schattierung ein/aus.
Die verfügbaren Tastaturkürzel sind oben aufgeführt:

- https://github.com/devemux86/cruiser/discussions/309#discussioncomment-11408233

**Comment by Basti32** [comment]
Danke für die schnelle Antwort @devemux86,
da mit den +/- und Pfeiltasten hatte ich nicht erwähnt, geht natürlich auch. Das geht sogar im Android + c für Standort, b für Favorit setzen, f für Standort folgen starten und n für Abos.

Aber in der oben genannten Aufzählung geht bei mir nur die Taste I im Desktop-Cruiser.
Oder mache ich was falsch?
Viele Grüße
Basti

Transaltor Deppl:

Thanks for the quick reply @devemux86, 
I hadn't mentioned the +/- and arrow keys, of course they work too. 
This even works in Android + c for location, b for set favourite, f for start location follow and n for subscriptions. 

But in the above list, only the I key works for me in the desktop cruiser. Or am I doing something wrong? 
Best regards 
Basti.

**Reply by devemux86** [reply]
@Basti32 

Route must have via or shaping points for the keys S, V, W to work (on the desktop).

Damit die Schlüssel S, V, W funktionieren, muss die Route Via- oder Shaping-Punkte haben (auf dem Desktop).

**Comment by Basti32** [comment]
Hallo @devemux86 ,
hatte einen Interpretationfehler. Es geht ja nicht um neue Punkte setzen sondern nur um das Ein- und Ausblenden dieser Punkte.

Viele Grüße
Basti

Transaltor DeepL

Hello @devemux86 , 
had an interpretation error. It is not about setting new points but only about showing and hiding these points. 

Best regards 
Basti

**Comment by pooatt** [comment]
Hi, 
I'm almost in the same setup as @Basti32 : An android device with a bluetooth keyboard and I succeed to use the following keys :

- `c `for current location.
- `b` for set favourite
- `f` for start location follow
- `n` for subscription
- `ê` for zoom in (yes that's strange, maybe because I use a bépo keyboard layout…)
- The arrows can move the map

But beside that I would like to know how to : 

- zoom-out 
- start a POI search

Thanks for your work.
Pooatt

**Reply by devemux86** [reply]
@pooatt 
Select "Keyboard" in "External input device" setting and can use `+` and `-` to zoom the map.

**Reply by devemux86** [reply]
@pooatt `N` starts / stops navigation, which is available in the subscription.

**Reply by pooatt** [reply]
> @pooatt Select "Keyboard" in "External input device" setting and can use `+` and `-` to zoom the map.

Thanks for your answer.
I already changed this setting. I continued to test and discovered that I can zoom-in with `=` and zoom-out with `%` 

> @pooatt `N` starts / stops navigation, which is available in the subscription.

Thanks, I'll subscribe asap.

**Reply by devemux86** [reply]
> I continued to test and discovered that I can zoom-in with `=` and zoom-out with `%`

@pooatt This depends on the keyboard, usually the `+` and `-` are together with other keys.

**Comment by Basti32** [comment]
Hallo @pooatt,
auf der deutschen Tastatur ist es die Tasten ß (=) Zoom - und ´ (%) Zoom + , und geht auch auf meiner Tastatur. Da ich zur Zeit kein Abo habe, verstehe ich jetzt auch die Funktion der N-Taste. Nur das die N-Taste bei der französischen Tastatur bei mit aber die Ö-Taste ist, trotzdem habe ich die gleiche Funktion bei meiner N-Taste. ê-Taste, ist bei mir die < Taste und ohne Funktion. 
Das ist sehr interessant. Das in manchen Fällen die Taste und Mahnmal das Zeichen/ Symbol zählt.

Viele Grüße
Basti

Transaltor DeepL:

Hello @pooatt, 
on the German keyboard it is the keys ß (=) Zoom - and ' (%) Zoom + , and also works on my keyboard. As I don't have a subscription at the moment, I now understand the function of the N key. The only difference is that the N key on my French keyboard is the Ö key, but I still have the same function with my N key. The ê key is the &lt; key for me and has no function, which is very interesting. That in some cases the key and reminder counts the character/symbol. 
best regards 
Basti
